Reinstate correctly functioning fp-error code for Linux; disable fp_error code for MacOS (will reinstate later)

jeffriley opened the pull request:
Reinstates the correct fp error handling code for Linux, and disables fp error handling for MacOS.  I will reinstate the MacOS code later when I figure out how to make it work properly for both INTEL and ARM architectures (it sort-of works for INTEL, but has some anomalies that need to be resolved; ARM will need some more work).

Silently ignores the `--fp-error-mode` option for MacOS.

jeffriley committed:
Reinstate correctly function fp-error code for Linux; disable fp_error code for MacOS (will reinstate later)

jeffriley committed:
Merge pull request #1189 from jeffriley/disable-fpe-macos

Reinstate correctly function fp-error code for Linux; disable fp_error code for MacOS (will reinstate later)

ilyamandel reviewed (commented):
First of all, apologies once again for my previous very crude changes that broke the functionality -- I just wanted a quick solution that would allow code to compile on all OS.  Thank you for these fixes.  I look forward to a Mac-friendly version, too. ;-)

On line 869, in the comment, "yes" seems slightly confusing (since it's a few lines away from the nearest if).  

Why is typecasting to void needed on lines 572 and 573 (but not 869 and 870)?

PRs #1186 and #1188 will need to be rolled in to this one, happy to approve once that's done.

jeffriley committed:
Update main.cpp

Removed extraneous comment (line)870
Added (void) to lines 869 & 870

jeffriley commented:
Thanks @ilyamandel 

> First of all, apologies once again for my previous very crude changes that broke the functionality -- I just wanted a quick
> solution that would allow code to compile on all OS.

No worries - understood.  I should have checked compilation on MacOS before I pushed the code...

> On line 869, in the comment, "yes" seems slightly confusing (since it's a few lines away from the nearest if).

Fixed (removed "yes" - it was a leftover from a copy&paste)

> Why is typecasting to void needed on lines 572 and 573 (but not 869 and 870)?

Technically casting the functional return to void is not required - the compiler is smart enough to handle it.  I like to add it explicitly so that future readers of the code know that I am deliberately discarding the return value, and that it's not a potential cause of failure.  I missed lies 869 and 870 -  I have now added "(void)" to those lines.

jeffriley committed:
Merge branch 'dev' into disable-fpe-macos
